### PR TITLE
feat: add --format json stdout output to eval, analyze, and perf commands

### LINKED ISSUE #847: feat: add --format json stdout output to winml eval
## Summary

Add `--format json` flag to `winml eval` to emit structured JSON to stdout, enabling Copilot agent skills to parse evaluation results programmatically.

## Motivation

Agent skills (`debug-accuracy-drop`, `validate-before-ship`, `optimize-for-device`) need to run `winml eval` and interpret output autonomously. Without machine-readable output, the agent must ask the user to paste results back manually.

`winml inspect` and `winml sys` already support `--format json` on stdout. This brings `winml eval` to parity.

## Proposed output schema

```json
{
  "mode": "compare",
  "model": "path/to/quantized.onnx",
  "model_id": "microsoft/resnet-50",
  "metrics": {
    "cosine_similarity": 0.87,
    "sqnr_db": 28.3,
    "psnr_db": 31.1,
    "max_abs_diff": 0.042
  },
  "task_metric": { "top1_accuracy": 0.741 },
  "threshold_pass": false
}
```

## Implementation notes

- Follow existing pattern in `commands/inspect.py` and `commands/sys.py`
- Banner/spinner output goes to stderr; JSON goes to stdout
- `--format` accepts `text` (default) or `json`
- File: `src/winml/modelkit/commands/eval.py`

## Related

Part of winml-cli skills agentic execution initiative. Enables: `debug-accuracy-drop`, `validate-before-ship`, `optimize-for-device` skills.

### LINKED ISSUE #848: feat: add --format json stdout output to winml analyze
## Summary

Add `--format json` flag to print structured operator coverage results to stdout (in addition to the existing `--output file.json` file-based output), enabling Copilot agent skills to parse compatibility results programmatically.

## Motivation

Agent skills (`ep-compatibility-check`, `debug-accuracy-drop`) need to run `winml analyze` and inspect EP operator support autonomously. The existing `--output file.json` requires writing to disk; agents need stdout JSON to avoid temp file management.

`winml inspect` and `winml sys` already support `--format json` on stdout. This brings `winml analyze` to parity.

## Proposed output schema

```json
{
  "ep": "qnn",
  "model": "path/to/model.onnx",
  "summary": { "supported": 142, "partial": 3, "unsupported": 1 },
  "partial_ops": ["MultiHeadAttention", "LayerNorm", "Softmax"],
  "unsupported_ops": ["CustomRotaryEmbedding"]
}
```

## Implementation notes

- `--output file.json` behavior unchanged
- Add `--format json` to also print the same data to stdout
- File: `src/winml/modelkit/commands/analyze.py`

## Related

Part of winml-cli skills agentic execution initiative. Enables: `ep-compatibility-check`, `debug-accuracy-drop` skills.

### LINKED ISSUE #849: feat: add --format json stdout output to winml perf
## Summary

Add `--format json` flag to print benchmark results to stdout (in addition to the existing `--output file.json` file-based output), enabling Copilot agent skills to parse latency results programmatically.

## Motivation

Agent skills (`optimize-for-device`, `validate-before-ship`) need to run `winml perf` and compare latency numbers across precision/EP configurations autonomously. The existing `--output file.json` requires writing to disk; agents need stdout JSON for clean pipeline chaining.

`winml inspect` and `winml sys` already support `--format json` on stdout. This brings `winml perf` to parity.

## Proposed output schema

```json
{
  "model": "path/to/model.onnx",
  "ep": "qnn",
  "device": "npu",
  "iterations": 100,
  "latency_ms": { "p50": 18.3, "p90": 21.7, "p99": 28.4, "mean": 18.9 },
  "throughput_rps": 54.6
}
```

## Implementation notes

- `--output file.json` behavior unchanged
- Add `--format json` to also print summary stats to stdout
- File: `src/winml/modelkit/commands/perf.py`

## Related

Part of winml-cli skills agentic execution initiative. Enables: `optimize-for-device`, `validate-before-ship` skills.

### PR DESCRIPTION
**Pull request by DingmaomaoBJTU**
## Summary

Add and unify `-f`/`--format` option across **all** data-producing commands, enabling programmatic JSON output for Copilot agent skills and scripting workflows.

- **New**: `winml eval`, `winml analyze`, `winml perf`, `winml catalog` now support `--format json`
- **Unified**: `winml inspect`, `winml sys`, `winml run` migrated to shared `format_option()` decorator
- Shared `OutputFormat` type alias and `cli_utils.format_option()` decorator eliminates duplicate option definitions

Closes #847, closes #848, closes #849
Ref #858

### New `--format json` support

| Command | `--format` choices | Default |
|---------|-------------------|---------|
| `eval` | `text`, `json` | `text` |
| `analyze` | `text`, `json` | `text` |
| `perf` | `text`, `json` | `text` |
| `catalog` | `text`, `json` | `text` |

### Unified to shared `format_option()` decorator

| Command | `--format` choices | Default | Notes |
|---------|-------------------|---------|-------|
| `inspect` | `table`, `json` | `table` | Was inline option |
| `sys` | `text`, `json`, `compact` | `text` | Was inline option |
| `run` | `text`, `json` | `text` | Was inline option; uses `--format` (no `-f` short flag, conflicts with `--file`) |

### Implementation details

- `cli_utils.format_option(choices, default, short_flag)` — shared Click decorator
- `OutputFormat` TypeAlias for static typing
- All banner/spinner/progress output routed to stderr in JSON mode
- JSON emitted via `click.echo(json.dumps(..., indent=2))` to stdout
- `analyze`: JSON emitted **before** `sys.exit()` so partial-support results (exit 1) still produce output

### `winml eval --format json`
```json
{
  "mode": "onnx",
  "model_id": "microsoft/resnet-50",
  "task": "image-classification",
  "device": "npu",
  "ep": "QNNExecutionProvider",
  "precision": "w8a16",
  "dataset": {
    "path": "timm/mini-imagenet",
    "samples": 100,
    "split": "test"
  },
  "metrics": {
    "top1_accuracy": 0.741,
    "top5_accuracy": 0.912
  }
}
```

### `winml analyze --format json`
```json
{
  "analysis_timestamp": "2026-06-10T10:30:00",
  "metadata": {
    "model_path": "model.onnx",
    "opset_version": 13,
    "total_operators": 146,
    "operator_counts": {"Conv": 53, "Add": 50, "Relu": 43},
    "unique_operator_types": 3
  },
  "results": [
    {
      "ep": "QNNExecutionProvider",
      "device": "NPU",
      "supported": 142,
      "partial": 3,
      "unsupported": 1,
      "partial_ops": ["MultiHeadAttention", "LayerNorm", "Softmax"],
      "unsupported_ops": ["CustomRotaryEmbedding"]
    }
  ]
}
```

### `winml perf --format json`
```json
{
  "benchmark_info": {
    "model_id": "microsoft/resnet-50",
    "device": "npu",
    "ep": "QNNExecutionProvider",
    "iterations": 100,
    "warmup": 10,
    "batch_size": 1
  },
  "latency_ms": {
    "mean": 18.9,
    "min": 16.2,
    "max": 28.4,
    "p50": 18.3,
    "p90": 21.7,
    "p95": 23.1,
    "p99": 28.4,
    "std": 2.1
  },
  "throughput": {
    "samples_per_sec": 54.6,
    "batches_per_sec": 54.6
  }
}
```

### `winml catalog --format json`
```json
[
  {
    "model_id": "google-bert/bert-base-uncased",
    "model_type": "bert",
    "size_mb": 104.4,
    "task": "fill-mask",
    "supported_eps": {
      "cpu": ["CPU"],
      "dml": ["GPU"],
      "qnn": ["GPU", "NPU"],
      "openvino": ["CPU", "GPU", "NPU"]
    }
  },
  {
    "model_id": "microsoft/resnet-50",
    "model_type": "resnet",
    "size_mb": 97.8,
    "task": "image-classification",
    "supported_eps": {
      "cpu": ["CPU"],
      "dml": ["GPU"],
      "qnn": ["GPU", "NPU"]
    }
  }
]
```

### `winml inspect --format json`
```json
{
  "model_path": "model.onnx",
  "opset_version": 13,
  "ir_version": 8,
  "producer": "pytorch 2.1",
  "inputs": [
    {"name": "pixel_values", "shape": [1, 3, 224, 224], "dtype": "float32"}
  ],
  "outputs": [
    {"name": "logits", "shape": [1, 1000], "dtype": "float32"}
  ],
  "operators": {"Conv": 53, "Add": 50, "Relu": 43}
}
```

### `winml sys --format json`
```json
{
  "platform": "Windows-11-10.0.26100",
  "python": "3.11.14",
  "onnxruntime": "1.21.1",
  "execution_providers": [
    {"name": "QNNExecutionProvider", "available": true},
    {"name": "DmlExecutionProvider", "available": true},
    {"name": "CPUExecutionProvider", "available": true}
  ],
  "devices": {
    "cpu": "13th Gen Intel Core i7-1365U",
    "gpu": "Intel Iris Xe Graphics",
    "npu": "Intel AI Boost"
  }
}
```

### `winml run --format json`
```json
{
  "model_id": "microsoft/resnet-50",
  "task": "image-classification",
  "ep": "QNNExecutionProvider",
  "outputs": {
    "logits": [[0.02, 0.01, 0.95, "..."]]
  },
  "postprocessed": {
    "label": "golden retriever",
    "score": 0.95
  }
}
```

## Design decisions

- `--format json` and `--output` are independently combinable
- JSON uses `indent=2` for readability, consistent across all commands
- All banner/spinner/progress output goes to stderr in JSON mode
- `--format` defaults to `text` (or `table` for inspect) to preserve backward compatibility
- `format_option(short_flag=False)` for `run` to avoid conflict with `-f`/`--file`

### CODE REVIEW

File: src/winml/modelkit/commands/analyze.py [hunk 1]
```diff
@@ -669,6 +670,14 @@ def _ep_name_device_display_name(ep_name: str, device_name: str) -> str:
     default=None,
     help="Path to HTP metadata JSON file for enhanced pattern extraction",
 )
+@click.option(
+    "-f",
+    "--format",
+    "output_format",
+    type=click.Choice(["text", "json"], case_sensitive=False),
```
Comment (xieofxie) [inline src/winml/modelkit/commands/analyze.py]: use type instead of str and also share the option

**Review by zhenchaoni — APPROVED** [review]

**Review by xieofxie — APPROVED** [review]

File: src/winml/modelkit/commands/sys.py [hunk 2]
```diff
@@ -666,14 +666,7 @@ def _output_ep_text(eps: list[dict[str, Any]]) -> None:
...
 @click.command()
-@click.option(
-    "--format",
-    "-f",
-    "output_format",
-    type=click.Choice(["text", "json", "compact"], case_sensitive=False),
-    default="text",
-    help="Output format: text (human-readable), json, or compact",
-)
+@cli_utils.format_option(choices=["text", "json", "compact"], default="text")
```
Comment (xieofxie) [inline src/winml/modelkit/commands/sys.py:669]: cli_utils.OutputFormat doesn't have compact

File: src/winml/modelkit/utils/cli.py [hunk 3]
```diff
@@ -124,6 +127,34 @@ def output_option(help_text: str, required: bool = False) -> Callable[[F], F]:
     return click.option("--output", "-o", **kwargs)
 
 
+def format_option(
+    choices: list[str] | None = None,
+    default: str = "text",
```
Comment (xieofxie) [inline src/winml/modelkit/utils/cli.py]: should be list[TypeAlias], TypeAlias

**Review by xieofxie — APPROVED** [review]

File: src/winml/modelkit/commands/sys.py
Comment (xieofxie) [inline src/winml/modelkit/commands/sys.py]: since output_format is typed, maybe we could remove all .lower() comparisons

**Review by xieofxie — APPROVED** [review]
